source/FreeRTOS_Sockets.c

```diff
@@ -4387,6 +4387,13 @@ void vSocketWakeUpUser( FreeRTOS_Socket_t * pxSocket )
         {
             pxBuffer = pxSocket->u.xTCP.txStream;
 
+            if( ( pxBuffer == NULL ) && ( pxSocket->u.xTCP.bits.bMallocError != pdTRUE ) )
+            {
+                /* Create the outgoing stream only when it is needed */
+                ( void ) prvTCPCreateStream( pxSocket, pdFALSE );
+                pxBuffer = pxSocket->u.xTCP.txStream;
+            }
```

Comment:
~~~diff
-    if( ( pxBuffer == NULL ) && ( pxSocket->u.xTCP.bits.bMallocError != pdTRUE ) )
+    if( ( pxBuffer == NULL ) && ( pxSocket->u.xTCP.bits.bMallocError != pdFALSE_UNSIGNED ) )
     {
         /* Create the outgoing stream only when it is needed */
         ( void ) prvTCPCreateStream( pxSocket, pdFALSE );
         pxBuffer = pxSocket->u.xTCP.txStream;
     }
~~~